docs/api_reference/src/quantities.tex

```diff
@@ -146,121 +464,1387 @@
+to specify conversions with other types representing quantities.
+\ref{qty.chrono} implements them for \tcode{std::chrono::duration} and \tcode{std::chrono::time_point}.
+
+\begin{itemdecl}
+template<typename T, template<typename> typename Traits>
+concept @\defexposconceptnc{qty-like}@ = requires(const T& qty, const Traits<T>::rep& num) {
+  requires !@\exposidnc{is-specialization-of}@(^T, ^quantity);
+  requires !@\exposidnc{is-specialization-of}@(^T, ^quantity_point);
+  { Traits<T>::to_numerical_value(qty) } -> std::@\stdconcept{same_as}@<typename Traits<T>::rep>;
+  { Traits<T>::from_numerical_value(num) } -> std::@\stdconcept{same_as}@<T>;
+  { Traits<T>::explicit_import } -> std::same_as<const bool>;
+  { Traits<T>::explicit_export } -> std::same_as<const bool>;
+  typename std::bool_constant<Traits<T>::explicit_import>;
+  typename std::bool_constant<Traits<T>::explicit_export>;
+};
+
+template<typename T>
+concept @\deflibconcept{quantity_like}@ =                 //
+  @\exposconceptnc{qty-like}@<T, quantity_like_traits> &&  //
+  requires {
+    typename quantity<quantity_like_traits<T>::reference, typename quantity_like_traits<T>::rep>;
+  };
+
+template<typename T>
+concept @\deflibconcept{quantity_point_like}@ =
+  @\exposconceptnc{qty-like}@<T, quantity_point_like_traits> &&  //
+  requires {
+    typename quantity_point<quantity_point_like_traits<T>::reference,
+                            typename quantity_point_like_traits<T>::point_origin,
+                            typename quantity_point_like_traits<T>::rep>;
+  };
+\end{itemdecl}
+
+\pnum
+In the following descriptions, let
+\begin{itemize}
+\item
+\tcode{Traits} be \tcode{quantity_like_traits} or \tcode{quantity_point_like_traits},
+\item
+\tcode{Q} be a type for which \tcode{Traits<Q>} is specialized,
+\item
+\tcode{qty} be an lvalue of type \tcode{const Q}, and
+\item
+\tcode{num} be an lvalue of type \tcode{const Tratis<Q>::rep}.
+\end{itemize}
+
+\pnum
+\tcode{Q} models \tcode{\exposconceptnc{qty-like}<Traits>} if and only if:
+\begin{itemize}
+\item
+\tcode{Traits<Q>::to_numerical_value(qty)} returns the numerical value\irefiev{112-01-29} of \tcode{qty}.
+\item
+\tcode{Traits<Q>::from_numerical_value(num)} returns a \tcode{Q} with numerical value \tcode{num}.
+\item
+If \tcode{Traits} is \tcode{quantity_point_like_traits},
+both numerical values are offset from \tcode{Traits<Q>::point_origin}.
+\end{itemize}
+
+\pnum
+If the following expression is \tcode{true}, the specified conversion will be explicit.
+\begin{itemize}
+\item
+\tcode{Traits<Q>::explicit_import} for the conversion from \tcode{Q} to a quantity type.
+\item
+\tcode{Traits<Q>::explicit_export} for the conversion from a quantity type to \tcode{Q}.
+\end{itemize}
+
+\rSec2[qty.types.general]{General}
+
+\pnum
+\label{term.quantity.type}
+A \defnadj{quantity}{type}
+is a type \tcode{\placeholder{Q}}
+that is a specialization of \tcode{quantity} or \tcode{quantity_point}.
```

Comment:
There are some users that derive from quantity already.  We support this in mp-units with derived_from. Should we remove this support?

Reply:
Also, I would not account quantity_point to a quantity type. One can do quantity equations on quantity types, which is not true for quantity points.

Putting them into the same basket makes it also harder to communicate.

Reply:
https://github.com/mpusz/mp-units/discussions/271
https://github.com/mpusz/mp-units/pull/286

Reply:
This is controversial for me as well. In C++ we typically discourage people from inheriting from `std` types. It would be great if we could find a better solution here.

Reply:
The most glaring issue is that for a parameter constrained with
`std::derived_from<quantity>` or `std::derived_from<quantity_point>`,
the implementation doesn't convert to the `quantity` base subobject.

For example, in `operator+`:
```C++
  template<std::derived_from<quantity> Q, auto R2, typename Rep2>
    requires detail::CommonlyInvocableQuantities<std::plus<>, quantity, quantity<R2, Rep2>>
  [[nodiscard]] friend constexpr Quantity auto operator+(const Q& lhs, const quantity<R2, Rep2>& rhs)
  {
    using ret = detail::common_quantity_for<std::plus<>, quantity, quantity<R2, Rep2>>;
    const ret ret_lhs(lhs);
```
We need this conversion from `Q`, type of `lhs`, to `ret`,
to be the same as `ret(static_cast<const quantity&>(lhs))`.
The implementation doesn't do that (like [_`as-variant`_](https://eel.is/c++draft/variant.visit)),
so we need wording to require those parameter types
to [not hide members](https://eel.is/c++draft/type.traits#meta.rqmts-1.sentence-4)
and prevent conversions like the above to have a different meaning.
Perhaps by requiring `Q` to behave as-if the base template in the specified wording
(e.g., by borrowing <https://eel.is/c++draft/namespace.std#2.2>).

Reply:
I do not oppose removing the support for deriving from a `quantity`.